libs/SalesforceAnalytics/SalesforceAnalytics/Classes/Store/SFSDKEventStoreManager.m

```diff
@@ -69,10 +69,12 @@ - (instancetype) initWithStoreDirectory:(NSString *) storeDirectory dataEncrypto
 
         // Gets current number of events stored.
         self.eventCountMutex = [[NSObject alloc] init];
-        self.numStoredEvents = 0;
-        NSArray *files = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:self.storeDirectory error:nil];
-        if (files) {
-            self.numStoredEvents = files.count;
+        @synchronized (self.eventCountMutex) {
```

Comment:
You just created the object you synchronize on at line 71, so how will it keep other threads out?
Should you do @synchronized ([self class])  instead?